test/python/quantum_info/channel/test_chi.py

```diff
@@ -0,0 +1,433 @@
+        mat16 = np.eye(16) / 4
+        chan = Chi(mat16)
+        self.assertAllClose(chan.data, mat16)
+        self.assertEqual(chan.dims, (4, 4))
+
+        # Wrong input or output dims should raise exception
+        self.assertRaises(QiskitError, Chi, mat16, input_dim=2, output_dim=4)
+
+        # Non multi-qubit dimensions should raise exception
+        self.assertRaises(
+            QiskitError, Chi, np.eye(6) / 2, input_dim=3, output_dim=2)
+
+    def test_equal(self):
+        """Test __eq__ method"""
+        mat = self.rand_matrix(4, 4, real=True)
+        self.assertEqual(Chi(mat), Chi(mat))
+
+    def test_copy(self):
+        """Test copy method"""
+        mat = np.eye(4)
+        orig = Chi(mat)
+        cpy = orig.copy()
+        cpy._data[0, 0] = 0.0
+        self.assertFalse(cpy == orig)
+
+    def test_evolve(self):
+        """Test evolve method."""
+        input_psi = [0, 1]
+        input_rho = [[0, 0], [0, 1]]
+
+        # Identity channel
+        chan = Chi(self.chiI)
+        target_rho = np.array([[0, 0], [0, 1]])
+        self.assertAllClose(chan.evolve(input_psi), target_rho)
+        self.assertAllClose(chan.evolve(np.array(input_psi)), target_rho)
+        self.assertAllClose(chan.evolve(input_rho), target_rho)
+        self.assertAllClose(chan.evolve(np.array(input_rho)), target_rho)
+
+        # Hadamard channel
+        chan = Chi(self.chiH)
+        target_rho = np.array([[1, -1], [-1, 1]]) / 2
+        self.assertAllClose(chan.evolve(input_psi), target_rho)
+        self.assertAllClose(chan.evolve(np.array(input_psi)), target_rho)
+        self.assertAllClose(chan.evolve(input_rho), target_rho)
+        self.assertAllClose(chan.evolve(np.array(input_rho)), target_rho)
+
+        # Completely depolarizing channel
+        chan = Chi(self.depol_chi(1))
+        target_rho = np.eye(2) / 2
+        self.assertAllClose(chan.evolve(input_psi), target_rho)
+        self.assertAllClose(chan.evolve(np.array(input_psi)), target_rho)
+        self.assertAllClose(chan.evolve(input_rho), target_rho)
+        self.assertAllClose(chan.evolve(np.array(input_rho)), target_rho)
+
+    def test_is_cptp(self):
+        """Test is_cptp method."""
+        self.assertTrue(Chi(self.depol_chi(0.25)).is_cptp())
+        # Non-CPTP should return false
+        self.assertFalse(
+            Chi(1.25 * self.chiI - 0.25 * self.depol_chi(1)).is_cptp())
+
+    def test_compose_except(self):
+        """Test compose different dimension exception"""
+        self.assertRaises(QiskitError, Chi(np.eye(4)).compose, Chi(np.eye(16)))
+        self.assertRaises(QiskitError, Chi(np.eye(4)).compose, np.eye(4))
+        self.assertRaises(QiskitError, Chi(np.eye(4)).compose, 2)
+
+    def test_compose(self):
+        """Test compose method."""
+        # Random input test state
+        rho = self.rand_rho(2)
+
+        # UnitaryChannel evolution
+        chan1 = Chi(self.chiX)
+        chan2 = Chi(self.chiY)
+        chan = chan1.compose(chan2)
```

Comment:
I not sure I really like this order

Reply:
We need this order to be consistent with the rest of the circuit gate type objects.